test/graphgym/test_graphgym.py

```diff
@@ -75,3 +76,51 @@ def test_run_single_graphgym():
 
     shutil.rmtree(cfg.out_dir)
     shutil.rmtree(cfg.dataset.dir)
+
+
+def trivial_metric(true, pred, task_type):
+    return {'trivial': 1}
+
+
+def test_run_trivial_graphgym():
```

Comment:
Hi @JiaxuanYou I looked a little bit more into the ``update_stats`` function and I don't think it will call the custom metrics directly. In my current setup, which is the same as that in the original setup, the ``task_stats`` (i.e. the metrics) are updated during the function call of ``write_epoch`` using ``self._train`` and ``self._pred`` as input, both of which are set during ``update_stats``.

So ``update_stats``does not directly call ``trivial`` but sets the ``self._train`` and ``self._pred`` instead. Let me know if I'm understanding this wrong or if you want me to handle the custom metric function calls differently.

Reply:
I think what @JiaxuanYou means is that we never actually test if `trivial_metric` is executed. To fix, we might want to add a counter to `trivial_metric`, similar to [here](https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/conv/test_message_passing.py#L195-L218).